collectfast/management/commands/collectstatic.py

```diff
@@ -32,13 +39,20 @@ def __init__(self, *args, **kwargs):
         self.tasks = []
         self.etags = {}
         self.collectfast_enabled = settings.enabled
-        if self.storage.preload_metadata is not True:
-            self.storage.preload_metadata = True
-            warnings.warn(
-                "Collectfast does not work properly without "
-                "`preload_metadata` set to `True` on the storage class. Try "
-                "setting `AWS_PRELOAD_METADATA` to `True`. Overriding "
-                "`storage.preload_metadata` and continuing.")
+        if settings.enabled:
+            if django_settings.STATICFILES_STORAGE not in VALID_BACKENDS:
```

Comment:
one problem with this is that it would cause a RuntimeError if you're using a subclass of S3BotoStorage/S3Boto3Storage. I see two other options:

(1) import the classes and use `if isinstance(self.storage, S3BotoStorage) or isinstance(self.storage, S3Boto3Storage)` (which could be refactored into a helper: `if is_s3_backend(self.storage)`)
(2) duck typing: `if not hasattr(self.storage, 'preload_metadata'): raise` --> but I'm wondering if we'd just want to set `preload_metadata` defensively in that case:
````python
if not getattr(self.storage, 'preload_metadata', False):
    self.storage.preload_metadata = True
    warnings.warn(...)
````

Reply:
Yes, that's a good point @pachewise.

Actually I wanted to implement it as in your first suggestion but encountered problems with mocked objects in some existing tests.

In any case, I think both of your suggestions could work, however looks like we have bigger questions now following @antonagestam's comment below. Let's clear the desired behavior first, and then I'll solve the problems in the implementation.